openmls/src/group/core_group/create_commit.rs

```diff
@@ -160,29 +162,43 @@ impl CoreGroup {
                 || contains_own_updates
                 || params.force_self_update()
             {
-                if params.commit_type() != CommitType::External {
+                let mut new_keypairs = if params.commit_type() != CommitType::External {
                     // If we're in the tree, we rekey our existing leaf.
                     let own_diff_leaf = diff
                         .own_leaf_mut()
                         .map_err(|_| LibraryError::custom("Unable to get own leaf from diff"))?;
-                    own_diff_leaf.rekey(
+                    let encryption_keypair = own_diff_leaf.rekey(
                         self.group_id(),
                         self.ciphersuite,
                         ProtocolVersion::default(), // XXX: openmls/openmls#1065
                         params.credential_bundle(),
                         backend,
                     )?;
-                }
+                    vec![encryption_keypair]
+                } else {
+                    // If we're not in the tree, we just load
+                    // KeyPackage we prepared earlier.
+                    let own_leaf = diff.own_leaf()
+                        // We should have a leaf in the diff
+                        .map_err(|_| LibraryError::custom("Unable to get own leaf from diff"))?;
+                    let encryption_keypair = EncryptionKeyPair::read_from_key_store(backend, own_leaf.encryption_key())
```

Comment:
Hm, this read  here isn't great.
It's not a public function and it's pretty deep in the code such that it will be difficult to split the function on the read.
If it's something we just created, we should be able to pass it around instead of storing it?
The key package creation above where this is coming from, has the same issue. Could we have the key package builder above in a way that it is returning the keys instead of storing them (at least as an option)? Then we wouldn't write there and wouldn't read here.